### ISSUE TITLE
cannot found cudatoolkit(12.1) when install it via conda in dockerfile.python

### ISSUE THREAD

**Issue by johnnyzhon** [NVIDIA/spark-rapids-ml#414]
=> ERROR [6/9] RUN conda install -y -c rapidsai -c conda-forge -c nvidia cudatoolkit=12.1 cuml=23.08     && c  393.8s
------
 > [6/9] RUN conda install -y -c rapidsai -c conda-forge -c nvidia cudatoolkit=12.1 cuml=23.08     && conda clean --all -f -y:
#5 1.045 Collecting package metadata (current_repodata.json): ...working... done
#5 73.24 Solving environment: ...working... unsuccessful initial attempt using frozen solve. Retrying with flexible solve.
#5 73.24 Collecting package metadata (repodata.json): ...working... done
#5 387.6 Solving environment: ...working... unsuccessful initial attempt using frozen solve. Retrying with flexible solve.
#5 387.6
#5 387.6 PackagesNotFoundError: The following packages are not available from current channels:
#5 387.6
#5 387.6   - cudatoolkit=12.1
#5 387.6
#5 387.6 Current channels:
#5 387.6
#5 387.6   - https://conda.anaconda.org/rapidsai/linux-64
#5 387.6   - https://conda.anaconda.org/rapidsai/noarch
#5 387.6   - https://conda.anaconda.org/conda-forge/linux-64
#5 387.6   - https://conda.anaconda.org/conda-forge/noarch
#5 387.6   - https://conda.anaconda.org/nvidia/linux-64
#5 387.6   - https://conda.anaconda.org/nvidia/noarch
#5 387.6   - https://repo.anaconda.com/pkgs/main/linux-64
#5 387.6   - https://repo.anaconda.com/pkgs/main/noarch
#5 387.6   - https://repo.anaconda.com/pkgs/r/linux-64
#5 387.6   - https://repo.anaconda.com/pkgs/r/noarch

**Comment by eordentlich** [comment]
According to the RAPIDS selector: https://docs.rapids.ai/install
try `conda install -y -c rapidsai -c conda-forge -c nvidia  cuml=23.08 cuda-version=12.0`

**Comment by johnnyzhon** [comment]
okay, thanks for your information.
Shall we going to modify the Dockerfile.python?

**Comment by eordentlich** [comment]
Yes.  We should switch to cuda-version from cudatoolkit.  Was already done in the ci/Dockerfile.

**Comment by johnnyzhon** [comment]
2895.6
2895.6 Package cuda-version conflicts for:
2895.6 cuda-version=12.1
2895.6 cuml=23.10 -> cuda-version[version='>=11,<12.0a0|>=12,<13.0a0|>=12.0,<13']
2895.6 cuml=23.10 -> cupy[version='>=12.0.0'] -> cuda-version[version='10.2.*|11.0.*|>=11.2,<12|11.1.*']The following specifications were found to be incompatible with your system:
2895.6
2895.6   - feature:/linux-64::__glibc==2.31=0
2895.6   - cuml=23.10 -> __glibc[version='>=2.17,<3.0.a0']
2895.6   - cuml=23.10 -> cupy[version='>=12.0.0'] -> __glibc[version='>=2.17']
2895.6   - python=3.9 -> libgcc-ng[version='>=10.3.0'] -> __glibc[version='>=2.17']
2895.6
2895.6 Your installed version is: 2.31
2895.6
2895.6

**Comment by johnnyzhon** [comment]
seems like cuml required cuda<12.0.  Is that a know limitation ?

**Comment by eordentlich** [comment]
cuml is compatible with cuda 12.0   cuda-version=12.0 should work in the conda install command.

**Comment by johnnyzhon** [comment]
Okay,  does cuml compatible with cuda 12.1+?  Is this requirment make sense still?

**Comment by lijinf2** [comment]
I think using cuda-version=12.0 may be more stable, as it is specified in cuml repository and tested: https://github.com/rapidsai/cuml/blob/branch-23.12/conda/environments/all_cuda-120_arch-x86_64.yaml

Is there any problem using cuda-version=12.0 in the conda install command?

**Comment by johnnyzhon** [comment]
using cuda-version=12.0 is OK, but i am wondering if we can use latest cuda version in future.
